Add Startness probes

xstefank opened the pull request:
Fixes #274

xstefank commented:
@pgunapal can you take a look please?

pgunapal commented on api/src/main/java/org/eclipse/microprofile/health/Startness.java:
```diff
@@ -0,0 +1,60 @@
+@Retention(RUNTIME)
+@Documented
+@Qualifier
+public @interface Startness {
```
I don't think the word `Startness` to define the new health check procedure make sense here. I think we should call it `Startup` to be consistent with K8s.

xstefank replied:
This is an intentional decision to mainly not introduce any confusion with https://javaee.github.io/javaee-spec/javadocs/javax/ejb/Startup.html as we are targeting Java EE developers. Also, it stays aligned with the other annotations. I understand this concern, it also took me a lot of thinking and internal discussions but I was convinced that Startness would be better. Please also note that this was introduced in our implementation a while back and I haven't heard any concerns from the community about the name.

pgunapal replied:
Right, that makes sense! Totally forgot about the EJB Startup bean. Since, the name is consistent with the existing health check procedures, I think we should be good.

xstefank replied:
Hello @tjquinno. Thank you for your feedback. I understand this is not grammatically correct but after weighing all arguments this came to us as the best alternative.

The discussions I mentioned happened in our implementation way before we proposed this functionality to the specification. This is why they were internal. Startness was provided in our implementation for several months now.

Personally, I still prefer Startness because the conflict with Startup is more important to me but this specification is a collaborative effort and it's not my place to discard any other opinions. However, as you've mentioned the plan ballot is ending today and I planned to release RC1 tomorrow. Going into this discussion now would most likely delay the release of Health 3.1 and in turn, also delay the release of MP 4.1. With this in mind, I would like to continue with the current naming and if we run into issues with the client adoption (note that I didn't receive this kind of feedback in our implementation up to this day) we will change it in Health 4.0 with most likely MP 6.x. Would this be acceptable?

tjquinno replied:
Of course, that's fine with me. Commenting at this point I never suggested nor did I expect the wheels of progress to stop turning on Health 3.1 or MP 4.1. Even so, I did want to explain my thinking, and I appreciate you accepting it in the collaborative spirit I intended.

Emily-Jiang replied:
@tjquinno at the moment, 3.1-RC1 was cut. However it is not too late to make more changes. This 3.1RC1 was released to get more feedback. As for this name, either `Startup` or `Startness` has drawbacks. How about `StartupCheck`? Though it is not same as startup probe but it avoids the name clashes with ejb startup and it is closer to the startup probe than the name `Startness`, which I also found difficult to comprehend.

xstefank replied:
Yes, we have reopened the naming discussion because there is a bigger pushback. But please keep your responses in the GG thread - https://groups.google.com/g/microprofile/c/j0f2zRBBMPU/m/LpQGlDqwAAAJ.

pgunapal commented on spec/src/main/asciidoc/protocol-wireformat.adoc:
```diff
@@ -212,10 +212,16 @@ Aspects regarding the secure access of health check information.
 | Readiness
 | See Appendix B
 
+| /health/start
```
In the protocol-wireformat doc, we should also update the `Health Check Procedures` section to mention the new `Startup` health check procedure.

pgunapal reviewed (changes requested):
The changes look good overall. Just that we should rename the new health check procedure to `Startup`, instead of `Startness` to be consistent with the k8s terminology. Also, we should update the README.md to mention the new health check procedure as well.

xstefank commented:
@pgunapal I've updated the README and the mentioned section to include startness. I also added a new config property, similar to empty readiness for startness. Can you please check again?
